lib/team.js

```diff
@@ -0,0 +1,93 @@
+let ChatMessage
+let MessageBuilder
+
+module.exports = loader
+
+function loader (mcVersion) {
+  ChatMessage = require('prismarine-chat')(mcVersion)
+  MessageBuilder = ChatMessage.MessageBuilder
+  return Team
+}
+
+function colorString (color) {
+  const formatting = [
+    'black',
+    'dark_blue',
+    'dark_green',
+    'dark_aqua',
+    'dark_red',
+    'dark_purple',
+    'gold',
+    'gray',
+    'dark_gray',
+    'blue',
+    'green',
+    'aqua',
+    'red',
+    'light_purple',
+    'yellow',
+    'white',
+    'obfuscated',
+    'bold',
+    'strikethrough',
+    'underlined',
+    'italic',
+    'reset'
+  ]
+  if (color === undefined || color > 21 || color === -1) return 'reset'
+  return formatting[color]
+}
+
+class Team {
+  constructor (packet) {
+    this.team = packet.team
+    this.update(packet)
+    this.membersMap = {}
+  }
+
+  parseMessage (value) {
+    let result
+    try {
+      result = new ChatMessage(JSON.parse(value)) // version>1.13-pre7
+    } catch {
+      result = MessageBuilder.fromString(value, { colorSeparator: '§' })
+      if (result === null) {
+        return new ChatMessage('')
+      }
+      return new ChatMessage(result.toJSON())
+    }
+    return result
+  }
+
+  add (name) {
+    this.membersMap[name] = ''
+    return this.membersMap[name]
+  }
+
+  remove (name) {
+    const removed = this.membersMap[name]
+    delete this.membersMap[name]
+    return removed
+  }
+
+  update (packet) {
```

Comment:
could you do the packet handling in the team plugin ?
we try to keep such classes independent from the packets so they stay fairly stable
(that means taking as input directly {name, friendlyFire, ...} here and explicitely passing each of those fields in the team plugin
that will make it easier to adapt when mojang (inevitably) changes the fields